src/main/java/org/desarrolladorslp/technovation/services/impl/DeliverableServiceImpl.java

```diff
@@ -26,6 +35,40 @@ public DeliverableDTO save(DeliverableDTO deliverableDTO) {
         return convertToDTO(deliverableRepository.save(deliverable));
     }
 
+    @Override
+    @Transactional(readOnly = true)
+    public List<DeliverableDTO> findByBatch(UUID batchId) {
+        Batch batch = new Batch();
+
+        batch.setId(batchId);
+        List<Deliverable> deliverables = deliverableRepository.findByBatch(batch);
+
+        return deliverables.stream().map(this::convertToDTO).collect(Collectors.toList());
+    }
+
+    @Override
+    @Transactional
+    public DeliverableDTO update(DeliverableDTO deliverableDTO, UUID deliverableId) {
+        deliverableDTO.setId(deliverableId);
+        Deliverable deliverable = convertToEntity(deliverableDTO);
+        return convertToDTO(deliverableRepository.save(deliverable));
+    }
+
+    @Override
+    @Transactional
+    public void delete(UUID deliverableId) {
+
+        try {
+            Optional<Deliverable> deliverableOptional = deliverableRepository.findById(deliverableId);
+            deliverableOptional.ifPresent(
+                    deliverable -> deliverableRepository.delete(deliverable)
+            );
+        } catch (Exception exception) {
+
+            logger.warn(exception.getMessage());
```

Comment:
In order to print exception traces, we must pass as the last parameter (after first) the exception as a Throwable instance... change it to:

```
logger.warn("Error trying to delete a deliverable with id {}", deliverableId, exception);
```